### DISCUSSION TITLE
Results of m1-machine-learning-test- M1 Macbook Air

### DISCUSSION THREAD

**Discussion by adityagupta1729** [mrdbourke/m1-machine-learning-test#12] [Q&A]
I tried running the process you explained and at the last, this is appearing in jupyter notebook, can you please help me here 

Thanks in advance!

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Input In [1], in <cell line: 4>()
      2 import pandas as pd
      3 import sklearn
----> 4 import tensorflow as tf
      5 import matplotlib.pyplot as plt
      7 # Check for TensorFlow GPU access

ModuleNotFoundError: No module named 'tensorflow'

**Comment by mrdbourke** [comment]
HI @adityagupta1729,

Did you try running the following lines *inside* your environment?

```
conda install -c apple tensorflow-deps
```
```
python -m pip install tensorflow-macos
```
```
python -m pip install tensorflow-metal
```

This should give you access to TensorFlow.

**Reply by adityagupta1729** [reply]
Thanks for the reply @mrdbourke. Actually, all the things got installed properly but there was a problem with the python path. Now that got resolved and I got stuck in the below-given error, any workarounds for this --

InvalidArgumentError: Cannot assign a device for operation sequential_1/random_flip/stateful_uniform_full_int/RngReadAndSkip: Could not satisfy explicit device specification '' because the node {{colocation_node sequential_1/random_flip/stateful_uniform_full_int/RngReadAndSkip}} was colocated with a group of nodes that required incompatible device '/job:localhost/replica:0/task:0/device:GPU:0'. All available devices [/job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:GPU:0]. Colocation Debug Info: Colocation group had the following types and supported devices: Root Member(assigned_device_name_index_=2 requested_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' assigned_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' resource_device_name_='/job:localhost/replica:0/task:0/device:GPU:0' supported_device_types_=[CPU] possible_devices_=[] RngReadAndSkip: CPU _Arg: GPU CPU

<img width="573" alt="Screenshot 2022-05-15 at 10 11 01 PM" src="https://user-images.githubusercontent.com/82268620/168484094-651044b1-151a-4f87-82c7-5a8e919aab0f.png">
